apps/contract-sync-job/src/main.ts

```diff
@@ -1 +1,183 @@
-console.log('Hello World!');
+import { GovrnContract, NetworkConfig } from '@govrn/govrn-contract-client';
+import { GovrnGraphClient } from '@govrn/govrn-subgraph-client';
+import { GovrnProtocol, SortOrder } from '@govrn/protocol-client';
+import { ethers } from 'ethers';
+import { GraphQLClient } from 'graphql-request';
+
+const protcolUrl = process.env.PROTOCOL_URL;
+const ENDPOINT = process.env.SUBGRAPH_URL;
+const jobName = 'contract-sync-job';
+
+const contractAddress = '0xefad47a6888762b666727cae19d5e502f0a68f43';
+const networkConfig: NetworkConfig = {
+  address: contractAddress,
+  chainId: 2,
+};
+
+const provider = new ethers.providers.JsonRpcProvider(
+  'https://rinkeby.infura.io/v3/cda40b6016f04aa48da096c096068718'
+);
+const govrnContract = new GovrnContract(networkConfig, provider);
+
+const getOrInsertUser = async (
+  govrn: GovrnProtocol,
+  data: { address: string }
+): Promise<number> => {
+  // Query existing user.
+  const user = await govrn.user.list({
+    where: {
+      address: { equals: data.address },
+    },
+    first: 1,
+  });
+
+  // Returns the id, if user exists.
+  if (user.length != 0) {
+    return user[0]?.id;
+  }
+
+  // Insert new user into db, if user doesn't exist.
+  const newUser = await govrn.user.create({
+    data: {
+      address: data.address,
+      chain_type: {
+        connectOrCreate: {
+          where: { name: 'example2' },
+          create: { name: 'example2' },
```

Comment:
Can we set this to rinkeby in this case? With a comment to pass in the appropriate chain?